tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -94,74 +89,79 @@ def test_upgrade_downgrade_streams(self, from_version, to_version):
         (search for get_kafka()). For streams in particular, that means that someone has manually
         copies the kafka-stream-$version-test.jar in the right S3 bucket as shown in base.sh.
         """
-        # Setup phase
-        self.zk = ZookeeperService(self.test_context, num_nodes=1)
-        self.zk.start()
-
-        # number of nodes needs to be >= 3 for the smoke test
-        self.kafka = KafkaService(self.test_context, num_nodes=3,
-                                  zk=self.zk, version=KafkaVersion(from_version), topics=self.topics)
-        self.kafka.start()
-        
-        # allow some time for topics to be created
-        time.sleep(10)
-        
-        self.driver = StreamsSmokeTestDriverService(self.test_context, self.kafka)
-        self.processor1 = StreamsSmokeTestJobRunnerService(self.test_context, self.kafka)
-
-        
-        self.driver.start()
-        self.processor1.start()
-        time.sleep(15)
+        if from_version != to_version:
+            # Setup phase
+            self.zk = ZookeeperService(self.test_context, num_nodes=1)
+            self.zk.start()
```

Comment:
I think you can do the zookeeper start in the `setUp` method

Reply:
hm. actually, there isn't a `setUp` method... Are you suggesting I make one?

FWIW, I think it's actually fine like this. The test is linearly readable this way.

Reply:
There's a `setUp` method on the [Test](https://github.com/confluentinc/ducktape/blob/master/ducktape/tests/test.py#L98) class that we can override.  It is fine like it is, but most of the other tests do ZK start-up in the `setUp` method, so I was aiming for consistency.

Reply:
aha, I gotcha. sure, I can do that.

Reply:
+1

Reply:
Ok, I'm glad that I know what you were talking about now. I actually think in this case, it's a little better as is, since we want to start the broker with different versions during the test.

We could still do it, but then we'd have to stop the broker and upgrade it and restart it during the test, or only start zk in startUp and start the broker during the tests... I think in this case, it's a more complicated control flow in the tests in exchange for diminishing returns.

I can totally get it for tests that just need zk and kafka running and don't care about the version, but in this case, I think I like it better flat.